### PR TITLE
Support fuse clean policy for JindoRuntime

### LINKED ISSUE #1254: [FEATURES]Support fuse clean policy for JindoRuntime.
**What feature you'd like to add:**
Support fuse clean policy for JindoRuntime. 

Fluid adopts fuse lazy start since v0.7.0. This PR supports two different fuse clean policy for JindoRuntime: `OnDemand` and `OnRuntimeDeleted`. `OnDemand` is the default clean policy and it cleans fuse pods whenever the fuse pod on some node is not in use, while with the `OnRuntimeDeleted` policy, once a fuse pod is lazily created, it will never be cleaned until the cache runtime is deleted. 

`OnDemand` is useful for saving cluster resources, while `OnRuntimeDeleted` is useful for avoiding frequent restarting.

User can specify the fuse clean policy by setting it in JindoRuntime:
```
apiVersion: data.fluid.io/v1alpha1
kind: JindoRuntime
metadata:
  name: hadoop
spec:
  ...
  fuse:
    # cleanPolicy: OnDemand
    cleanPolicy: OnRuntimeDeleted
```

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
Support fuse clean policy for JindoRuntime. 

Fluid adopts fuse lazy start since v0.7.0. This PR supports two different fuse clean policy for JindoRuntime: `OnDemand` and `OnRuntimeDeleted`. `OnDemand` is the default clean policy and it cleans fuse pods whenever the fuse pod on some node is not in use, while with the `OnRuntimeDeleted` policy, once a fuse pod is lazily created, it will never be cleaned until the cache runtime is deleted. 

`OnDemand` is useful for saving cluster resources, while `OnRuntimeDeleted` is useful for avoiding frequent restarting.

User can specify the fuse clean policy by setting it in JindoRuntime:
```
apiVersion: data.fluid.io/v1alpha1
kind: JindoRuntime
metadata:
  name: hadoop
spec:
  ...
  fuse:
    # cleanPolicy: OnDemand
    cleanPolicy: OnRuntimeDeleted
```

**Note:** For now, fuse lazy start is only supported for JindoRuntime, that's why this PR considers JindoRuntime only. Fuse clean policy feature will also be supported for other runtime once the fuse lazy start feature is supported.

### Ⅱ. Does this pull request fix one issue?

fixes #1254 

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
- pkg/ddc/base/runtime_test.go

### Ⅳ. Describe how to verify it
1. Create the following Dataset and JindoRuntime with different `spec.fuse.cleanPolicy`
```
kind: Dataset
metadata:
  name: hadoop
spec:
  mounts:
    - mountPoint: oss://tz-fluid-datasets/
      options:
        fs.oss.accessKeyId: <OSS_ACCESS_KEY_ID>
        fs.oss.accessKeySecret: <OSS_ACCESS_KEY_SECRET>
        fs.oss.endpoint: <OSS_ENDPOINT>
      name: hadoop
---
apiVersion: data.fluid.io/v1alpha1
kind: JindoRuntime
metadata:
  name: hadoop
spec:
  replicas: 1
  tieredstore:
    levels:
      - mediumtype: HDD
        path: /mnt/disk1
        quota: 5Gi
        high: "0.9"
        low: "0.8"
  fuse:
    cleanPolicy: OnRuntimeDeleted
```
2. Create and delete Pods with related PVC. Watch the lifecycle of fuse pod

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/fluid/fluid/crds/data.fluid.io_databackups.yaml [hunk 1]
```diff
@@ -1,13 +1,32 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (cheyang) [inline charts/fluid/fluid/crds/data.fluid.io_databackups.yaml:3]: You need to upgrade controller-gen version. Currently Fluid is on CRD v1.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: charts/fluid/fluid/crds/data.fluid.io_databackups.yaml [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.7.0
+    controller-gen.kubebuilder.io/version: v0.3.0
```
Comment (cheyang) [inline charts/fluid/fluid/crds/data.fluid.io_databackups.yaml:7]: controller-gen version should be 0.7.0.

File: charts/fluid/fluid/crds/data.fluid.io_databackups.yaml (hunk 1 above)
Comment (TrafalgarZZZ) [reply charts/fluid/fluid/crds/data.fluid.io_databackups.yaml]: Thx. Fixed.

File: charts/fluid/fluid/crds/data.fluid.io_databackups.yaml (hunk 2 above)
Comment (TrafalgarZZZ) [reply charts/fluid/fluid/crds/data.fluid.io_databackups.yaml]: Thx. Fixed.

**Review by cheyang — APPROVED** [review]

**Pull request merged**